traffic_ops/testing/api/v4/cdn_dnsseckeys_test.go

```diff
@@ -0,0 +1,156 @@
+package v4
+
+/*
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License.
+*/
+
+import (
+	"net/http"
+	"reflect"
+	"strconv"
+	"strings"
+	"testing"
+
+	tc "github.com/apache/trafficcontrol/lib/go-tc"
+	"github.com/apache/trafficcontrol/lib/go-util"
+	"github.com/apache/trafficcontrol/traffic_ops/testing/api/assert"
+	client "github.com/apache/trafficcontrol/traffic_ops/v4-client"
+)
+
+func TestCDNsDNSSEC(t *testing.T) {
+	if !includeSystemTests {
+		t.Skip()
+	}
+	WithObjs(t, []TCObj{CDNs, Types, Tenants, Users, Parameters, Profiles, Statuses, Divisions, Regions, PhysLocations, CacheGroups, Servers, Topologies, ServerCapabilities, ServiceCategories, DeliveryServices}, func() {
+		t.Run("GENERATE DNSSEC KEYS", func(t *testing.T) { GenerateDNSSECKeys(t) })
+		t.Run("REFRESH DNSSEC KEYS", func(t *testing.T) { RefreshDNSSECKeys(t) }) // NOTE: testing refresh last (while no keys exist) because it's asynchronous and might affect other tests
+	})
+}
+
+func RefreshDNSSECKeys(t *testing.T) {
+	resp, reqInf, err := TOSession.RefreshDNSSECKeys(client.RequestOptions{})
+	assert.NoError(t, err, "Unable to refresh DNSSEC keys: %v - alerts: %+v", err, resp.Alerts)
+	assert.Equal(t, reqInf.StatusCode, http.StatusAccepted, "Refreshing DNSSEC keys - Expected: status code %d, Actual: %d", http.StatusAccepted, reqInf.StatusCode)
+
+	loc := reqInf.RespHeaders.Get("Location")
+	if loc == "" {
+		t.Fatalf("Refreshing DNSSEC keys - Expected: non-empty 'Location' response header, Actual: empty")
+	}
+	asyncID, err := strconv.Atoi(strings.Split(loc, "/")[4])
```

Comment:
This should check if the string actually can be separated into 5 parts by a delimiter of `/` before doing this indexing, to avoid a segmentation fault.